### PR TITLE
[GLUTEN-7014][CH] Fix: different results from `get_json_object`

### PR DESCRIPTION
**Pull request by lgbo-ustc**
## What changes were proposed in this pull request?

(Please fill in changes proposed in this fix)

Fixes: #7014

Fix following case
```sql
select get_json_object('{"a":"\\\"b", "x":"u0001}Uûu0010Jw\\\"<洫C"}\0', '$.a');
```

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)

unit tests

(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

### CODE REVIEW

**Comment by lgbo-ustc** [comment]
Run Gluten Clickhouse CI

File: cpp-ch/local-engine/Functions/SparkFunctionGetJsonObject.h [hunk 1]
```diff
@@ -66,6 +68,263 @@ struct GetJsonObject
...
+ 
+        pos = normalizeWhitespace(pos, end, dst);
+
+        bool has_more = false;
+        while (pos && pos < end && *pos != ']')
+        {
+            has_more = false;
+            switch(*pos)
+            {
+                case '{': {
```
Comment (liuneng1994) [inline cpp-ch/local-engine/Functions/SparkFunctionGetJsonObject.h:232]: need format code

File: cpp-ch/local-engine/Functions/SparkFunctionGetJsonObject.h [hunk 2]
```diff
@@ -66,6 +68,261 @@ struct GetJsonObject
...
+        {
+            // LOG_DEBUG(getLogger("GetJsonObject"), "xxx normalizeArray. not ]");
+            return nullptr;
+        }
+        pos += 1;
+        copyToDst(dst, ']');
+        return normalizeWhitespace(pos, end, dst);
+    }
+
+    inline static const char * normalizeObject(const char * pos, const char * end, char *& dst)
```
Comment (liuneng1994) [inline cpp-ch/local-engine/Functions/SparkFunctionGetJsonObject.h:264]: 这里inline感觉没啥作用，有递归调用

Comment (lgbo-ustc) [reply cpp-ch/local-engine/Functions/SparkFunctionGetJsonObject.h:264]: 只是个提示，倒没什么影响

File: cpp-ch/local-engine/Functions/SparkFunctionGetJsonObject.h [hunk 3]
```diff
@@ -16,8 +16,8 @@
  */
 #pragma once
 #include <memory>
-#include <string_view>
 #include <stack>
```
Comment (KevinyhZou) [inline cpp-ch/local-engine/Functions/SparkFunctionGetJsonObject.h:19]: remove `#include <stack>`,  not used now

File: gluten-ut/spark32/src/test/scala/org/apache/gluten/utils/clickhouse/ClickHouseTestSettings.scala [hunk 4]
```diff
@@ -371,8 +371,6 @@ class ClickHouseTestSettings extends BackendTestSettings {
   enableSuite[GlutenIntervalFunctionsSuite]
   enableSuite[GlutenJoinSuite]
   enableSuite[GlutenJsonFunctionsSuite]
```
Comment (exmy) [inline gluten-ut/spark32/src/test/scala/org/apache/gluten/utils/clickhouse/ClickHouseTestSettings.scala:373]: The `JsonExpressionsSuite` of vanilla Spark ut seems missing in spark33 ut module, please add this suite to ensure cover more cases.

**Review by exmy — APPROVED** [review]

**Pull request merged**